datafusion/physical-plan/src/joins/utils.rs

```diff
@@ -1411,6 +1424,63 @@ where
         .collect::<PrimitiveArray<T>>()
 }
 
+/// Appends probe indices in order by considering the given build indices.
+///
+/// This function constructs new build and probe indices by iterating through
+/// the provided indices, and appends any missing values between previous and
+/// current probe index with a corresponding null build index. It handles various
+/// edge cases and returns an error if either index is `None`.
```

Comment:
It doesn't return `Result`.

```suggestion
/// current probe index with a corresponding null build index.
```